src/webui/utils/api.js

```diff
@@ -1,49 +1,84 @@
 import storage from './storage';
+import {HTTP_STATUS} from '../../../src/lib/constants';
 
-class API {
-  request(url, method = 'GET', options = {}) {
-      if (!window.VERDACCIO_API_URL) {
-        throw new Error('VERDACCIO_API_URL is not defined!');
-      }
+/**
+ * CHECK FOR BASE API URL
+ */
+if (!window.VERDACCIO_API_URL) {
+  throw new Error('VERDACCIO_API_URL is not defined!');
+}
+const BASE_URL = window.VERDACCIO_API_URL;
 
-      const token = storage.getItem('token');
-      if (token) {
-        if (!options.headers) options.headers = {};
+/**
+ * Handles response according to content type
+ * @param {object} response
+ * @returns {promise}
+ */
+function handleResponseType(response) {
+  if (response.headers) {
+    const contentType = response.headers.get('Content-Type');
+    if (contentType.includes('application/json')) {
+      return Promise.all([response.ok, response.json()]);
+    }
+    // it includes all text types
+    if (contentType.includes('text/')) {
+      return Promise.all([response.ok, response.text()]);
+    }
+  }
+};
 
-        options.headers.authorization = `Bearer ${token}`;
-      }
+/**
+ * checkAuthentication if 401 then clears storage
+ */
+function checkAuthentication(response) {
+  if (response.status === HTTP_STATUS.UNAUTHORIZED) {
+    storage.removeItem('token');
+    storage.removeItem('username');
+    /**
+     * @todo 
+     * Remove it when global storage available (e.g. Redux) which can re-render
+     * application without reloading it forcefully.
+     */
+    window.location.reload();
+  }
+  return response;
+}
 
-      if (!['http://', 'https://', '//'].some((prefix) => url.startsWith(prefix))) {
-        url = window.VERDACCIO_API_URL + url;
-      }
+/**
+ * API
+ */
+class API {
+  constructor(baseUrl) {
+    this.baseUrl = baseUrl;
+  }
 
-      /**
-       * Handles response according to content type
-       * @param {object} response
-       * @returns {promise}
-       */
-      function handleResponseType(response) {
-        if (response.headers) {
-          const contentType = response.headers.get('Content-Type');
-          if (contentType.includes('application/pdf')) {
-            return Promise.all([response.ok, response.blob()]);
-          }
-          if (contentType.includes('application/json')) {
-            return Promise.all([response.ok, response.json()]);
-          }
-          // it includes all text types
-          if (contentType.includes('text/')) {
-            return Promise.all([response.ok, response.text()]);
-          }
-        }
+  request(resource = '', method = 'GET', options = {}) {
+    let endpoint;
+    
+    // checks for token
+    const token = storage.getItem('token');
+    if (token) {
+      if (!options.headers) {
+        options.headers = {};
       }
+      options.headers.authorization = `Bearer ${token}`;
```

Comment:
`Bearer` already exist as constant `TOKEN_BEARER`

We have a utility to build this ``Bearer ${token}`` find for
```
export function buildToken(type: string, token: string) {
  return `${_.capitalize(type)} ${token}`;
}
```